contrib/example-backend/controllers/serviceexport/serviceexport_controller.go

```diff
@@ -64,219 +68,119 @@ func NewController(
+//+kubebuilder:rbac:groups=kubebind.k8s.io,resources=apiserviceexports,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=kubebind.k8s.io,resources=apiserviceexports/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=kubebind.k8s.io,resources=apiserviceexports/finalizers,verbs=update
+//+kubebuilder:rbac:groups=apiextensions.k8s.io,resources=customresourcedefinitions,verbs=get;list;watch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+func (r *APIServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling APIServiceExport", "namespace", req.Namespace, "name", req.Name)
+
+	// Fetch the APIServiceExport instance
+	apiServiceExport := &kubebindv1alpha2.APIServiceExport{}
+	if err := r.Get(ctx, req.NamespacedName, apiServiceExport); err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			logger.Info("APIServiceExport not found, ignoring")
+			return ctrl.Result{}, nil
 		}
-		logger.V(2).Info("queueing APIServiceExport", "key", key, "reason", "CustomResourceDefinition", "CustomResourceDefinitionKey", crdKey)
-		c.queue.Add(key)
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, fmt.Errorf("failed to get APIServiceExport: %w", err)
 	}
-}
-
-// Start starts the controller, which stops when ctx.Done() is closed.
-func (c *Controller) Start(ctx context.Context, numThreads int) {
-	defer runtime.HandleCrash()
-	defer c.queue.ShutDown()
-
-	logger := klog.FromContext(ctx).WithValues("controller", controllerName)
-
-	logger.Info("Starting controller")
-	defer logger.Info("Shutting down controller")
-
-	for i := 0; i < numThreads; i++ {
-		go wait.UntilWithContext(ctx, c.startWorker, time.Second)
-	}
-
-	<-ctx.Done()
-}
 
-func (c *Controller) startWorker(ctx context.Context) {
-	defer runtime.HandleCrash()
+	// Create a copy to modify
+	original := apiServiceExport.DeepCopy()
 
-	for c.processNextWorkItem(ctx) {
+	// Run the reconciliation logic
+	if err := r.reconciler.reconcile(ctx, apiServiceExport); err != nil {
+		logger.Error(err, "Failed to reconcile APIServiceExport")
+		return ctrl.Result{}, err
 	}
-}
 
-func (c *Controller) processNextWorkItem(ctx context.Context) bool {
-	// Wait until there is a new item in the working queue
-	key, quit := c.queue.Get()
-	if quit {
-		return false
+	// Update status if it has changed
+	if !reflect.DeepEqual(original, apiServiceExport) {
+		err := r.Update(ctx, apiServiceExport)
+		if err != nil {
+			logger.Error(err, "Failed to update APIServiceExport status")
+			return ctrl.Result{}, fmt.Errorf("failed to update APIServiceExport status: %w", err)
+		}
+		logger.Info("APIServiceExport status updated", "namespace", apiServiceExport.Namespace, "name", apiServiceExport.Name)
```

Comment:
This comment makes me wonder if you intended to update  the status subresource. But your code above says `r.Update()`, not `r.Status().Update()`.

Reply:
might be :) lets see :)